### PR TITLE
#2441: Improved caching of the rust inserts

### LINKED ISSUE #2441: Improve rust caching mechanism
It is much better to compare in `BinarizeMojo` sources of current rust insert with cached and avoid compilation if they are equal.

### PR DESCRIPTION
**Pull request by levBagryansky**
Closes #2441

---

### Detailed summary
- Added a new test method `boostsSecondCompilation` in the `BinarizeMojoTest` class.
- Added imports for `TextOf` and `UncheckedText` in the `BinarizeMojo` class.
- Added a new method `sameProject` in the `BinarizeMojo` class to check if the project was not changed.
- Added a new method `sameFile` in the `BinarizeMojo` class to check if the source file is the same as in cache.
- Added a new method `uncomment` in the `BinarizeMojo` class to remove the first line from a string.
- Updated the `build` method in the `BinarizeMojo` class to use the new methods and improve caching logic.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by levBagryansky** [comment]
@Graur please review

File: eo-maven-plugin/src/main/java/org/eolang/maven/BinarizeMojo.java [hunk 1]
```diff
@@ -147,31 +149,102 @@ private void build(final File project) throws IOException {
...
-                    .directory(project)
-            )
-        ) {
-            proc.stdout();
-        } catch (final IllegalArgumentException exc) {
-            throw new BuildFailureException(
+        if (BinarizeMojo.sameProject(
+            project.toPath(), this.cache
+            .resolve("Lib")
+            .resolve(project.getName())
```
Comment (Graur) [inline eo-maven-plugin/src/main/java/org/eolang/maven/BinarizeMojo.java:155]: @levBagryansky Pretty strange formatting for me. Can you please move `this.cache` to the new line?

File: eo-maven-plugin/src/main/java/org/eolang/maven/BinarizeMojo.java [hunk 2]
```diff
@@ -147,31 +149,102 @@ private void build(final File project) throws IOException {
...
+            );
+        } else {
+            Logger.info(
+                this,
                 String.format(
-                    "Failed to build cargo project with dest = %s",
-                    project
-                ),
-                exc
+                    "Building %s rust project..",
```
Comment (Graur) [inline eo-maven-plugin/src/main/java/org/eolang/maven/BinarizeMojo.java:168]: @levBagryansky Double dots? IT is a typo or you put some meaning here?

File: eo-maven-plugin/src/main/java/org/eolang/maven/BinarizeMojo.java [hunk 3]
```diff
@@ -147,31 +149,102 @@ private void build(final File project) throws IOException {
...
+                    project.getName()
+                )
+            );
+            try (
+                VerboseProcess proc = new VerboseProcess(
```
Comment (Graur) [inline eo-maven-plugin/src/main/java/org/eolang/maven/BinarizeMojo.java:173]: @levBagryansky Could be final?

File: eo-maven-plugin/src/main/java/org/eolang/maven/BinarizeMojo.java [hunk 4]
```diff
@@ -147,31 +149,102 @@ private void build(final File project) throws IOException {
...
+     * Check if the source file is the same as in cache.
+     * @param src Source file.
+     * @param cached Cache file.
+     * @return True if the same.
+     */
+    private static boolean sameFile(final Path src, final Path cached) {
+        return cached.toFile().exists() && BinarizeMojo.uncomment(
+            new UncheckedText(
+                new TextOf(
+                    src
```
Comment (Graur) [inline eo-maven-plugin/src/main/java/org/eolang/maven/BinarizeMojo.java:226]: @levBagryansky Do we really need this variable on the separate line?

File: eo-maven-plugin/src/main/java/org/eolang/maven/BinarizeMojo.java [hunk 5]
```diff
@@ -147,31 +149,102 @@ private void build(final File project) throws IOException {
...
+                )
+            ).asString()
+        ).equals(
+            new UncheckedText(
+                new TextOf(
+                    cached
```
Comment (Graur) [inline eo-maven-plugin/src/main/java/org/eolang/maven/BinarizeMojo.java:232]: @levBagryansky Same here. Do we really need this variable on the separate line?

**Review by Graur — CHANGES_REQUESTED** [review]
@levBagryansky Good one! Just a few comments below:

File: eo-maven-plugin/src/main/java/org/eolang/maven/BinarizeMojo.java (hunk 2 above)
Comment (levBagryansky) [reply eo-maven-plugin/src/main/java/org/eolang/maven/BinarizeMojo.java:169]: @Graur Yes. This is a long process with the following comments then.

File: eo-maven-plugin/src/main/java/org/eolang/maven/BinarizeMojo.java (hunk 3 above)
Comment (levBagryansky) [reply eo-maven-plugin/src/main/java/org/eolang/maven/BinarizeMojo.java]: @Graur PMD does not accept it:
```
[INFO] PMD: eo-maven-plugin/src/main/java/org/eolang/maven/BinarizeMojo.java[174-177]: Unnecessary modifier 'final' on resource specification 'proc': resource specifications are implicitly final (UnnecessaryModifier)
[INFO] Read our quality policy: http://www.qulice.com/quality.html

```

**Review by Graur — APPROVED** [review]
@levBagryansky Looks much better to me, thanks!
@yegor256 Please have a look

**Comment by levBagryansky** [comment]
@Graur could you please recheck it please

**Review by Graur — APPROVED** [review]

**Comment by levBagryansky** [comment]
@yegor256 please check

**Comment by levBagryansky** [comment]
@yegor256 please check

**Comment by yegor256** [comment]
@rultor merge

**Pull request merged**